Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2619 -- Add Sign In/Out to Header

Resolves [EREGCSC-2619](https://jiraent.cms.gov/browse/EREGCSC-2619)

**Description**

To improve usability of authentication in eRegs, we want to add sign-in and sign-out to our header navigation.

**This pull request changes:**

- Creates `useLoginRedirectUrl` [composable](https://vuejs.org/guide/reusability/composables) to ensure the redirect link to be used after a successful login is always correct no matter where the user is on the site
   - This will include query params and the URL hash
- Creates `useDropdownMenu` composable to encapsulate showing/hiding dropdown menu state logic
- Removes LoginIndicator at bottom right of screen that used to contain Log Out button
- Adds `sign-in` slot to `Header` component
   - When logged out, this slot will contain the `SignInLink` Vue component
   - When logged in, this slow will contain the `HeaderUserWidget` Vue component
- Creates new `HeaderUserWidget` component that includes a User Account SVG icon and an associated dropdown menu
   - Dropdown menu includes bolded username, some info, and a Sign Out button
   - Sign Out button will be the `oidc_logout` form when in Django template pages
   - Sign Out button will be a button that, when clicks, programmatically submits a hidden `oidc_logout` form on Vue Single Page App pages (Search, Statutes, Subjects)
- Adds associates styles
- Fixes broken e2e tests; creates new test suite to validate login and logout flows on different parts of the app

**Steps to manually verify this change:**

1. Green check marks
2. Visit [experimental deployment](https://c00mzmd0th.execute-api.us-east-1.amazonaws.com/dev1279)
3. Login and log out from the homepage
4. Login and log out from a SPA page ([subjects](https://c00mzmd0th.execute-api.us-east-1.amazonaws.com/dev1279/subjects/), for example)
5. When logging in on the Subjects page, select a subject and make sure that subject is still selected when redirected after login
